### ISSUE TITLE
Use FastAPI instead of Bottle for the internal server

### ISSUE THREAD

**Issue by Hedde** [ICTU/quality-time#490]
Replacing Bottle with [FastAPI](https://github.com/tiangolo/fastapi) might be an interesting tech improvement, because it will allow us to lift the features of python 3.6+ (e.g. asyncio) and it's type declaration support.

Blocked by #2961.

We're going to remove the internal server altogether. See #5789, #5790, and #5791.

**Comment by eric-nieuwland** [comment]
You may also want to have a look at Sanic - https://sanic.readthedocs.io

**Comment by Hedde** [comment]
Ref: https://fastapi.tiangolo.com/alternatives/#sanic

**Comment by fniessink** [comment]
Assumption: only the collector uses the internal API and there are two endpoints. The collector and notifier also use and keep using the public API (e.g. to get the data model). 

New component name: `internal-server`. The `server` component name remains unchanged.

The collector needs extra environment variables for the internal API component.

Remove `internal/` from the endpoints.

Use FastAPI instead of Bottle.

Update software docs, including components.dot.

Note: This results in a new major version of Quality-time since the composition changes.

Out of scope: 
- making the collector and notifier talk to the internal-server exclusively.
- simpler version of the `datamodel` endpoint to retrieve the latest version of the data model only.

**Comment by fniessink** [comment]
Blocked by #2960.
